chronos/asyncmacro2.nim

```diff
@@ -297,6 +297,11 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
 template await*[T](f: Future[T]): untyped =
   when declared(chronosInternalRetFuture):
     chronosInternalRetFuture.internalChild = f
+
+    when chronosFuturesInstrumentation:
```

Comment:
Why would this trigger a pause, its actually yielding execution to the child, not pausing it.

Reply:
It’s pausing direct execution of this future. This allows tracking the execution time of just this future.

Also it may be a while before the child future is scheduled and executed so pausing would still be needed. We track total exec time of the future and its children by passing both to the callback which can sum them as desired.

Ben made a diagram somewhere I think, describing the different time segments.